### PR TITLE
[Nara] polkadot-js/api update

### PR DESCRIPTION
**Pull request by mnaamani**
Bump polkadot-js/api to version v10.1.4, compatible with nara substrate version v0.9.41 (based on https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-40/2468)

Fixed some code that broke, and patched hydra-typegen to resolve typegen issues in query-node mappings https://github.com/Joystream/hydra/pull/537

closes https://github.com/Joystream/joystream/issues/4903 https://github.com/Joystream/joystream/issues/4928

### CODE REVIEW

**Review by kdembler — COMMENTED** [review]
I'm getting those resolutions warnings when running `yarn`:
```
warning Resolution field "@polkadot/api@10.1.4" is incompatible with requested version "@polkadot/api@^10.7.1"
warning Resolution field "typescript@5.0.2" is incompatible with requested version "typescript@^4.4"
warning Resolution field "typescript@5.0.2" is incompatible with requested version "typescript@^3.9.5"
warning Resolution field "typescript@5.0.2" is incompatible with requested version "typescript@2.2.2"
warning Resolution field "typescript@5.0.2" is incompatible with requested version "typescript@^4.4"
warning Resolution field "typescript@5.0.2" is incompatible with requested version "typescript@^3.9.7"
warning Resolution field "typescript@5.0.2" is incompatible with requested version "typescript@^3.8.3"
warning Resolution field "typescript@5.0.2" is incompatible with requested version "typescript@^3.0.3"
```

Not sure where those are coming from, seems all packages are already using typescript@5

File: joystreamjs/package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
   "dependencies": {
```
Comment (kdembler) [inline joystreamjs/package.json:43]: We should also bump joystreamjs package version

File: cli/package.json [hunk 2]
```diff
@@ -17,13 +17,13 @@
...
     "@oclif/plugin-not-found": "^1.2.4",
     "@oclif/plugin-warn-if-update-available": "^1.7.0",
-    "@polkadot/api": "8.9.1",
-    "@polkadot/keyring": "9.5.1",
-    "@polkadot/util": "9.5.1",
-    "@polkadot/util-crypto": "9.5.1",
-    "@substrate/txwrapper-core": "3.1.3",
-    "@substrate/txwrapper-substrate": "3.1.3",
-    "@substrate/txwrapper-polkadot": "3.1.3",
+    "@polkadot/api": "10.1.4",
```
Comment (kdembler) [inline cli/package.json:20]: Let's bump cli version as well

File: query-node/build.sh [hunk 3]
```diff
@@ -27,6 +27,7 @@ yarn
 ln -s ../../../../../node_modules/typeorm/cli.js ./generated/graphql-server/node_modules/.bin/typeorm
 
 yarn workspace query-node codegen
+cp graphql-server-tsconfig.json generated/graphql-server/tsconfig.json
```
Comment (mnaamani) [inline query-node/build.sh:30]: We should update the hydra package to add the change in the tsconfig.json there

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]
Looks good.

1. You need to typecast the `specVersion` in [members_MemberInvited](https://github.com/Joystream/joystream/blob/c93ce9a09a8cadb3da688970cc0d1b7f52d6282e/query-node/mappings/src/membership.ts#L453) mapping too.
2. After running `yarn build` the root `yarn.lock` file was updated with the following entry. Seems like after updating the dependency in `@joystream/js`, `yarn build` was not run
```
"@joystream/metadata-protobuf@^2.14.0":
  version "2.14.0"
  resolved "https://registry.yarnpkg.com/@joystream/metadata-protobuf/-/metadata-protobuf-2.14.0.tgz#83fa0297fb9b6e5e4224500a2c68d7714f273248"
  integrity sha512-1/FgUW6mGt3t+p0k7/KdKnRQPU0HghY4MJTSkip1SCTMICZWoDuSADJuM8nlh2Qdu9kqvodqtndKjz057b0t3g==
  dependencies:
    "@types/iso-3166-2" "^1.0.0"
    "@types/long" "^4.0.1"
    google-protobuf "^3.14.0"
    i18n-iso-countries "^6.8.0"
    iso-3166-2 "^1.0.0"
    iso-639-1 "^2.1.9"
    long "^4.0.0"
    protobufjs "^6.11.2"
"
```

Also I would say that just keep the same package version for `@joystream/metadata-protobuf` in all of the dependants, e.g. currently `@joystream/js`, `@joystream/distributor-cli` and `storage-node` all have different version for metadata-protobuf

**Review by zeeshanakram3 — APPROVED** [review]
LGTM.

**Comment by mnaamani** [comment]
> warning Resolution field "@polkadot/api@10.1.4" is incompatible with requested version "@polkadot/api@^10.7.1"

`@substrate/txwrapper` v6.0.1 packages are requesting v10.7.1 ... it wasn't a problem when using yarn, but now It has actually caused a problem after I published the joystream-cli package.. when running the published version / installed globally you see a bunch of warnings..

```
@polkadot/util-crypto has multiple versions, ensure that there is only one installed.
@polkadot/util has multiple versions, ensure that there is only one installed.

**Pull request merged**